tests/test_custom_attributes.py

```diff
@@ -61,9 +61,38 @@
 }
 
 
+class MiqAPIMock(Mock):
+    def __init__(self, *args, **kwargs):
+        super(MiqAPIMock, self).__init__(*args, **kwargs)
+        self.next_id = 1
+
+    def get_new_id(self, action):
+        if action != 'edit':
+            self.next_id = self.next_id + 1
+        return self.next_id
```

Comment:
When the action is edit, this returns the last assigned id, which may not be the c.a. you're editing at the moment?  It just happens to work because the update test updates EXISTING_CA which happens to have id 1.
I'd expect `post()` to do the conditional and just preserve id for `edit`.

P.S. python has `+= 1`